pkg/apis/validation/validation_test.go

```diff
@@ -35,83 +37,97 @@ func Test_ValidateAttributes(t *testing.T) {
 	}{
 		"attributes with no issuer name but DNS names should error": {
 			attr: map[string]string{
-				csiapi.CAFileKey:      "ca.crt",
-				csiapi.CertFileKey:    "crt.tls",
-				csiapi.KeyFileKey:     "key.tls",
-				csiapi.DNSNamesKey:    "foo.bar.com,car.bar.com",
-				csiapi.KeyEncodingKey: "PKCS1",
+				csiapi.CAFileKey:       "ca.crt",
+				csiapi.CertFileKey:     "crt.tls",
+				csiapi.KeyFileKey:      "key.tls",
+				csiapi.DNSNamesKey:     "foo.bar.com,car.bar.com",
+				csiapi.KeyEncodingKey:  "PKCS1",
+				csiapi.KeyAlgorithmKey: "RSA",
+				csiapi.KeySizeKey:      "2048",
 			},
 			expErr: field.ErrorList{
 				field.Required(field.NewPath("volumeAttributes", "csi.cert-manager.io/issuer-name"), "issuer-name is a required field"),
 			},
 		},
 		"attributes with common name but no issuer name or DNS names should error": {
 			attr: map[string]string{
-				csiapi.CAFileKey:      "ca.crt",
-				csiapi.CertFileKey:    "crt.tls",
-				csiapi.KeyFileKey:     "key.tls",
-				csiapi.CommonNameKey:  "foo.bar",
-				csiapi.KeyEncodingKey: "PKCS1",
+				csiapi.CAFileKey:       "ca.crt",
+				csiapi.CertFileKey:     "crt.tls",
+				csiapi.KeyFileKey:      "key.tls",
+				csiapi.CommonNameKey:   "foo.bar",
+				csiapi.KeyEncodingKey:  "PKCS1",
+				csiapi.KeyAlgorithmKey: "RSA",
+				csiapi.KeySizeKey:      "2048",
 			},
 			expErr: field.ErrorList{
 				field.Required(field.NewPath("volumeAttributes", "csi.cert-manager.io/issuer-name"), "issuer-name is a required field"),
 			},
 		},
 		"valid attributes with common name should return no error": {
 			attr: map[string]string{
-				csiapi.IssuerNameKey:  "test-issuer",
-				csiapi.CAFileKey:      "ca.crt",
-				csiapi.CertFileKey:    "crt.tls",
-				csiapi.KeyFileKey:     "key.tls",
-				csiapi.CommonNameKey:  "foo.bar",
-				csiapi.KeyEncodingKey: "PKCS1",
+				csiapi.IssuerNameKey:   "test-issuer",
+				csiapi.CAFileKey:       "ca.crt",
+				csiapi.CertFileKey:     "crt.tls",
+				csiapi.KeyFileKey:      "key.tls",
+				csiapi.CommonNameKey:   "foo.bar",
+				csiapi.KeyEncodingKey:  "PKCS1",
+				csiapi.KeyAlgorithmKey: "RSA",
+				csiapi.KeySizeKey:      "2048",
 			},
 			expErr: nil,
 		},
 		"valid attributes with DNS names should return no error": {
 			attr: map[string]string{
-				csiapi.IssuerNameKey:  "test-issuer",
-				csiapi.CAFileKey:      "ca.crt",
-				csiapi.CertFileKey:    "crt.tls",
-				csiapi.KeyFileKey:     "key.tls",
-				csiapi.DNSNamesKey:    "foo.bar.com,car.bar.com",
-				csiapi.KeyEncodingKey: "PKCS1",
+				csiapi.IssuerNameKey:   "test-issuer",
+				csiapi.CAFileKey:       "ca.crt",
+				csiapi.CertFileKey:     "crt.tls",
+				csiapi.KeyFileKey:      "key.tls",
+				csiapi.DNSNamesKey:     "foo.bar.com,car.bar.com",
+				csiapi.KeyEncodingKey:  "PKCS1",
+				csiapi.KeyAlgorithmKey: "RSA",
+				csiapi.KeySizeKey:      "2048",
```

Comment:
comment: It's so awesome to see good tests getting written for PRs like this - thank you, it makes it so much easier to review! ⭐